### PR TITLE
Embed doc URLs in swagger document

### LINKED ISSUE #3059: Embed doc URLs in swagger documentation
We should embed links to github docs in the API documentation. I think there's a swagger field for that.

/cc @nikhiljindal

### PR DESCRIPTION
**Pull request by janetkuo**
Add doc url to descriptions of swagger fields. e.g., 

type Container struct {
Image string '.... description:...**; see http://releases.k8s.io/HEAD/docs/Images.md**' 
}

Fixes #3059 

cc @bgrant0607 @nikhiljindal @thockin

### CODE REVIEW

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -178,35 +178,35 @@ type VolumeSource struct {
...
 	// ISCSI represents an ISCSI Disk resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
 	ISCSI *ISCSIVolumeSource `json:"iscsi,omitempty" description:"iSCSI disk attached to host machine on demand"`
 	// Glusterfs represents a Glusterfs mount on the host that shares a pod's lifetime
-	Glusterfs *GlusterfsVolumeSource `json:"glusterfs,omitempty" description:"Glusterfs volume that will be mounted on the host machine "`
+	Glusterfs *GlusterfsVolumeSource `json:"glusterfs,omitempty" description:"Glusterfs volume that will be mounted on the host machine; see http://releases.k8s.io/HEAD/examples/glusterfs/README.md"`
 	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
-	PersistentVolumeClaim *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty" description:"a reference to a PersistentVolumeClaim in the same namespace"`
+	PersistentVolumeClaim *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty" description:"a reference to a PersistentVolumeClaim in the same namespace; see http://releases.k8s.io/HEAD/docs/persistent-storage.md#tldr"`
```
Comment (nikhiljindal) [inline pkg/api/v1/types.go]: This doc link doesnt work

File: pkg/api/v1/types.go [hunk 2]
```diff
@@ -217,18 +217,18 @@ type PersistentVolumeClaimVolumeSource struct {
...
+	HostPath *HostPathVolumeSource `json:"hostPath,omitempty" description:"a HostPath provisioned by a developer or tester; for develment use only; see http://releases.k8s.io/HEAD/docs/volumes.md#hostpath"`
 	// Glusterfs represents a Glusterfs volume that is attached to a host and exposed to the pod
-	Glusterfs *GlusterfsVolumeSource `json:"glusterfs,omitempty" description:"Glusterfs volume resource provisioned by an admin"`
+	Glusterfs *GlusterfsVolumeSource `json:"glusterfs,omitempty" description:"Glusterfs volume resource provisioned by an admin; see http://releases.k8s.io/HEAD/examples/glusterfs/README.md"`
 	// NFS represents an NFS mount on the host
-	NFS *NFSVolumeSource `json:"nfs,omitempty" description:"NFS volume resource provisioned by an admin"`
+	NFS *NFSVolumeSource `json:"nfs,omitempty" description:"NFS volume resource provisioned by an admin; see http://releases.k8s.io/HEAD/docs/volumes.md#nfs"`
 	// RBD represents a Rados Block Device mount on the host that shares a pod's lifetime
 	RBD *RBDVolumeSource `json:"rbd,omitempty" description:"rados block volume that will be mounted on the host machine"`
```
Comment (nikhiljindal) [inline pkg/api/v1/types.go]: Why not add http://releases.k8s.io/HEAD/examples/rbd/README.md here?

File: pkg/api/v1/types.go [hunk 3]
```diff
@@ -178,35 +178,35 @@ type VolumeSource struct {
...
 	GitRepo *GitRepoVolumeSource `json:"gitRepo,omitempty" description:"git repository at a particular revision"`
 	// Secret represents a secret that should populate this volume.
-	Secret *SecretVolumeSource `json:"secret,omitempty" description:"secret to populate volume"`
+	Secret *SecretVolumeSource `json:"secret,omitempty" description:"secret to populate volume; see http://releases.k8s.io/HEAD/docs/volumes.md#secrets"`
 	// NFS represents an NFS mount on the host that shares a pod's lifetime
-	NFS *NFSVolumeSource `json:"nfs,omitempty" description:"NFS volume that will be mounted in the host machine"`
+	NFS *NFSVolumeSource `json:"nfs,omitempty" description:"NFS volume that will be mounted in the host machine; see http://releases.k8s.io/HEAD/docs/volumes.md#nfs"`
 	// ISCSI represents an ISCSI Disk resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
 	ISCSI *ISCSIVolumeSource `json:"iscsi,omitempty" description:"iSCSI disk attached to host machine on demand"`
```
Comment (nikhiljindal) [inline pkg/api/v1/types.go]: Link to examples/iscsi/README.md here?

**Comment by nikhiljindal** [comment]
There is a PR out to revert https://github.com/GoogleCloudPlatform/kubernetes/pull/10690. 
We will have to revert this as well if we go ahead with that.

cc @zmerlynn @thockin

File: pkg/api/v1/types.go (hunk 1 above)
Comment (JanetKuo) [inline pkg/api/v1/types.go]: Fixed. Use this doc link instead http://releases.k8s.io/HEAD/docs/persistent-volumes.md#persistentvolumeclaims

File: pkg/api/v1/types.go (hunk 2 above)
Comment (JanetKuo) [inline pkg/api/v1/types.go]: Fixed

File: pkg/api/v1/types.go (hunk 3 above)
Comment (JanetKuo) [inline pkg/api/v1/types.go]: Fixed

**Comment by nikhiljindal** [comment]
A lot of these docs dont add much value actually. But I guess, we will be improving our docs over time.
LGTM for now.

**Comment by JanetKuo** [comment]
@nikhiljindal ok to merge?

**Comment by nikhiljindal** [comment]
Yes. Since this is a doc only change, a single LGTM should be enough to merge.

**Pull request merged**